### PR TITLE
ETFE-3032 Message inbox sort functionality

### PR DESCRIPTION
**Pull request by grahampaulcook**

### CODE REVIEW

**Comment by grahampaulcook** [comment]
Button is missing when JavaScript is disabled

File: conf/messages [hunk 1]
```diff
@@ -317,13 +317,14 @@ viewAllMovements.sort.newest = Dispatched (newest)
...
 viewAllMessages.sort.label = Sort by
+viewAllMessages.sort.button = Sort messages
 viewAllMessages.sort.messageTypeA = Message (A - Z)
 viewAllMessages.sort.messageTypeD = Message (Z - A)
-viewAllMessages.sort.dateReceivedA = Received (newest)
-viewAllMessages.sort.dateReceivedD = Received (oldest)
+viewAllMessages.sort.dateReceivedD = Received (newest)
+viewAllMessages.sort.dateReceivedA = Received (oldest)
 viewAllMessages.sort.arcA = Arc (A - Z)
 viewAllMessages.sort.arcD = Arc (Z - A)
```
Comment (LewisMitchell) [inline conf/messages:329]: Unrelated to what you've done, but should Arc be captialised?

File: app/views/messages/ViewAllMessages.scala.html [hunk 2]
```diff
@@ -54,17 +54,27 @@
...
+                    classes = "govuk-!-display-inline govuk-!-padding-right-2"
+                ),
+                classes = "govuk-!-display-inline"
+            ))
+
+            @govukButton(
+                ButtonViewModel(Text(messages("viewAllMessages.sort.button")))
+                        .withCssClass("govuk-!-margin-left-2")
+                        .withId("sortBySubmit")
+            )
```
Comment (LewisMitchell) [inline app/views/messages/ViewAllMessages.scala.html:73]: Should the button be inline to the sort by dropdown?
image

File: conf/messages (hunk 1 above)
Comment (LewisMitchell) [reply conf/messages:329]: Sorting by ARC doesn't seem to be in the acceptance criteria?

File: app/views/messages/ViewAllMessages.scala.html (hunk 2 above)
Comment (LewisMitchell) [reply app/views/messages/ViewAllMessages.scala.html:73]: I think there be a test in the ViewSpec also?

**Review by LewisMitchell — CHANGES_REQUESTED** [review]

File: app/views/messages/ViewAllMessages.scala.html (hunk 2 above)
Comment (grahampaulcook) [reply app/views/messages/ViewAllMessages.scala.html]: Updated to align that button 
image

File: conf/messages (hunk 1 above)
Comment (grahampaulcook) [reply conf/messages:329]: > Sorting by ARC doesn't seem to be in the acceptance criteria?

The **messages-FrontendAPISpecification-MDTPtoEIS** document has only 4 possible sort values [messagetype, datereceived, arc & readindicator]

Comment (LewisMitchell) [reply conf/messages:329]: I think that was the intention, probably worth checking with UCD?

**Review by LewisMitchell — COMMENTED** [review]
Just one minor comment for UCD

**Pull request merged**